FEAT: Commons IP schema validation

carlwilson opened the pull request:
- removed expansions of status enums; and
- added a `ruleId` alias for validation `rule_id`.

carlwilson committed:
FEAT: Commons IP schema validation

- removed expansions of status enums; and
- added a `ruleId` alias for validation `rule_id`.

dockmd reviewed (approved):
LGTM, however why did you remove spaces from well and not well formed constants?

carlwilson commented:
> LGTM, however why did you remove spaces from well and not well formed constants?

That's because they're incompatible with commons-ip which doesn't include spaces. I need compatibility to auto compare the results via a JSON deserialisation. Once we have compatibility we can talk about changes, but I can't change commons-ip.

carlwilson committed:
Merge remote-tracking branch 'origin/integration' into fix/commons-ip

carlwilson committed:
MAINT: Merge integration into current branch.

carlwilson committed:
FIX: Deserialisation of rule_id

- added a `model_config` paramter to allow both `rule_id` and `ruleId` as keys in Pydantic model validation;
- fixes issue with deserialisation of `rule_id` in `Rule` model.

carlwilson committed:
Merge branch 'integration' into fix/commons-ip

carlwilson committed:
FEAT: Convert commons-ip representations

- added a `model_validator` to the `PackageDetails` class to convert incoming JSON dictionary to a `List` for now;
- added PyDantic config to `Result` to allow multiple names for `rule_id` during validation;
- reverted change to dictionary as it was not necessary;
- added tests and data for deserialisation of commons-ip types; and
- fixed minor compiler warnings.

carlwilson committed:
Merge branch 'fix/commons-ip' of github.com:E-ARK-Software/eark-validator into fix/commons-ip

carlwilson committed:
FEAT: Final commons-ip compatibility tweaks

- refactored `MetadataResults` to match `commons-ip`, it's probably better as well;
- moved `name` from `InformationPackage` to `PackageDetails` class;
- renamed `InformationPackage.package` to `InformationPackage.details`;
- renamed existing `ValidationReport.convert_dict` validator to `ValidationReport.convert_representations_dict` (more explicit);
- added a second validator, `VaidationReport.convert_checksum_ids`, to convert `commons-ip` checksum ids to `eark_validator` hyphenated form;
- `is_valid` convenience property to `ValidationReport` class;
- string constants for 'VALID' and 'INVALID'; and
- fixed tests to accomodate.

carlwilson commented:
Final time I promise @dockmd, this is now compatible with commons-ip JSON and the new validator rest service works.

dockmd commented on eark_validator/structure.py:
```diff
@@ -40,6 +39,7 @@
 
 METS_NAME = 'METS.xml'
 STR_REQ_PREFIX = 'CSIPSTR'
+ROOT = 'root '
```
Perhaps it is better to use ROOT without the space at the end and then combine it with f‘{ROOT} {self.name}’

carlwilson committed:
FIX: Support commons-ip use of NOTVALID

- added `convert_status` validator to `MetadataResults` class to convert commons-ip `NOTVALID` status to `INVALID`;
- moved checksum algorithm ID validation to the `PackageDetails` class;
- added test and test data for status conversion; and
- fixed type hinting for validation methods to `Any`.

carlwilson commented:
@dockmd, I should have kept my mouth shut. Found another change to make to coerce a final value. I'll take a look at your suggestion  about root and get a final commit up shortly. Looking to close this and merge to integrations ASAP as I can then get a test package out and an updated REST validator.

carlwilson committed:
REV: Tidier use of ROOT in structure.py

As suggested in this [review comment](https://github.com/E-ARK-Software/eark-validator/pull/60#discussion_r1735205478), thanks.
